### ISSUE TITLE
App.UnhandledException on XamarinFormsNav.PopAsync(true)

### ISSUE THREAD

**Issue by sjorsmiltenburg** [dotnet/maui#18467]
### Description

I have a semi-reproducable blocking bug in my main app where, when I navigate back on certain views and I get a 

UnhandledException with Microsoft.UI.Xaml.Controls.Frame.NavigationFailed was unhandled.
Inner Exception Message = "Catastrophic failure (0x8000FFFF (E_UNEXPECTED))"

caught in the App.UnhandledExceptionEventHandler
However afterwards it also hits my catch block where I see the exception as a

"Error HRESULT E_FAIL has been returned from a call to a COM component" when calling XamarinFormsNav.PopAsync(true);

with the call stack mentioned below

If I swallow this exception there is still a view on the navstack, but visually I end up in an empty looking view (blank page with no controls)

### Steps to Reproduce

I'm unable to pinpoint what exactly is causing this.

### Version with bug

8.0.0-rc.2.9373

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Did you find any workaround?

no

### Relevant log output

```shell
at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
   at ABI.Microsoft.UI.Xaml.Controls.IContentPresenterMethods.set_Content(IObjectReference _obj, Object value)
   at Microsoft.UI.Xaml.Controls.ContentPresenter.set_Content(Object value)
   at Microsoft.Maui.Platform.StackNavigationManager.OnNavigated(Object sender, NavigationEventArgs e)
   at WinRT._EventSource_global__Microsoft_UI_Xaml_Navigation_NavigatedEventHandler.EventState.<GetEventInvoke>b__1_0(Object sender, NavigationEventArgs e)
   at ABI.Microsoft.UI.Xaml.Navigation.NavigatedEventHandler.Do_Abi_Invoke(IntPtr thisPtr, IntPtr sender, IntPtr e)
--- End of stack trace from previous location ---
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
   at ABI.Microsoft.UI.Xaml.Controls.IFrameMethods.GoBack(IObjectReference _obj, NavigationTransitionInfo transitionInfoOverride)
   at Microsoft.UI.Xaml.Controls.Frame.GoBack(NavigationTransitionInfo transitionInfoOverride)
   at Microsoft.Maui.Platform.StackNavigationManager.NavigateTo(NavigationRequest args)
   at Microsoft.Maui.Handlers.NavigationViewHandler.RequestNavigation(INavigationViewHandler arg1, IStackNavigation arg2, Object arg3)
   at Microsoft.Maui.CommandMapper.InvokeCore(String key, IElementHandler viewHandler, IElement virtualView, Object args)
   at Microsoft.Maui.Handlers.ElementHandler.Invoke(String command, Object args)
   at Microsoft.Maui.Controls.NavigationPage.Microsoft.Maui.IStackNavigation.RequestNavigation(NavigationRequest eventArgs)
   at Microsoft.Maui.Controls.NavigationPage.<SendHandlerUpdateAsync>d__100.MoveNext()
   at Microsoft.Maui.Controls.NavigationPage.MauiNavigationImpl.<OnPopAsync>d__7.MoveNext()
   at BB.Maui.Services.NavService.<<GoBack>b__12_0>d.MoveNext() in C:\_Sources\BookaBooka\BB.Maui\Services\NavService.cs:line 79
```

**Comment by samfoucart** [comment]
I am encountering a similar issue when I use `Application.Current.MainPage.ShowPopup(new MyView())` from the Community Toolkit. If I swallow the exception, on Windows I get a blank page that looks like this.
![image](https://github.com/dotnet/maui/assets/43978545/df6e067e-da22-4ca1-abd0-b3c3cf90e07f)

It always happens on a specific view in my application, but I cannot figure out what is unique about this view for me to make a minimum reproducible sample. When I try to isolate it, the bug stops happening. But when I put the view back into the application, it happens 100% of the time.

**Comment by samfoucart** [comment]
@bengiles I figured out a solution to my problem btw. I forgot to ever update my comment. 

My issue specifically happens when you get to a page by calling `Navigation.PushModalAsync`. Our application was using the dependency `FreshMvvm.Maui`. We migrated this app from Xamarin, and it was using that dependency to handle navigation, but this should be reproducible without that dependency. As long as you're on a page inheriting from NavigationPage, and push a modal, if you then call `Application.Current.MainPage.ShowPopup(...)`, you will get this exception. This exception happens because the `Application.Current.MainPage` is not the actual page you're on if you're in a Modal.

To solve this, we did 
``` csharp
_loadingPopup = new LoadingPopup(message ?? " ");
if (Application.Current != null)
{
    if (Application.Current.MainPage != null)
    {
        Page mainPage = Application.Current.MainPage;
        if (mainPage.Navigation.ModalStack.Count == 0)
        {
            mainPage.ShowPopup(_loadingPopup);
        }
        else
        {
            IReadOnlyList<Page> stack = mainPage.Navigation.ModalStack;
            stack[stack.Count - 1].ShowPopup(_loadingPopup);
        }
    }
}
```

My issue was specific to the Popups from the Maui community toolkit, but I'm guessing that doing most actions to `Application.Current.MainPage` while you're in a modal will cause this exception.
